File: Problem4.java

class Problem4 {
    public static void main(String[] argv) {

        long max = 0;

        for(int i = 100; i < 999; i++) {
            for(int j = 100; j < 999; j++) {
                int sum = i * j;
                if(isPalindrome(sum) && sum > max) {
                    max = sum; 
                }
            }
        }

        System.out.println(max);

    }

    static boolean isPalindrome(int question) {
         int reverseNum = 0;
         int input = question;

         while(input != 0) {
            reverseNum = reverseNum * 10 + input % 10;
            input = input / 10;
         }

        return question == reverseNum;

    }
}

File: Problem4.rs

fn is_palindrome(question: i32) -> bool {
    let mut reverse_num = 0;
    let mut input = question;

    while input != 0 {
        reverse_num = reverse_num * 10 + input % 10;
        input = input / 10;
    }

    return question == reverse_num;
}

fn main() {
    let mut max: i32 = 0;
    let mut sum;

    for i in 100..999 {
        for j in 100..999 {
            sum = i * j;
            if is_palindrome(sum) && sum > max {
                max = sum; 
            }
        } 
    }

    println!("Sum: {}", max)
}
